CMakeLists.txt

```diff
@@ -213,6 +213,13 @@ if(OpenCL_FOUND)
   endif()
 endif()
 
+find_package(amgcl)
+if(amgcl_FOUND)
+  set(HAVE_AMGCL 1)
+  include_directories(${AMGCL_INCLUDE_DIRS})
+  set_source_files_properties(opm/simulators/linalg/bda/amgclSolverBackend.cpp PROPERTIES LANGUAGE CUDA)
```

Comment:
How is this working without CUDA? 
I am getting this error when activating only OpenCL, maybe that is related?
```
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_CUDA_COMPILE_WHOLE_COMPILATION
```